### PR TITLE
Feature: enable Azure PostgreSQL Server testing

### PR DESCRIPTION
**Pull request by omichels**

### CODE REVIEW

**Comment by rguthriemsft** [comment]
@HadwaAbdelhalem PTAL

**Review by HadwaAbdelhalem — CHANGES_REQUESTED** [review]
Thank you @omichels for your contribution. I left few comments on the PR/ also would you please execute the CI pipeline in your fork. #https://github.com/gruntwork-io/terratest/blob/master/.github/workflows/ci.yml.

File: examples/azure/terraform-azure-postgresql-example/main.tf [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  administrator_login          = "pgsqladmin"
+  administrator_login_password = "H@Sh1CoR3!"
+  version                      = "11"
+  ssl_enforcement_enabled      = true
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY AZURE PostgreSQL Database
+# ---------------------------------------------------------------------------------------------------------------------
+resource "azurerm_postgresql_database" "postgresqldb" {
```
[the remark this answers is not in the document]
Comment (omichels) [reply examples/azure/terraform-azure-postgresql-example/main.tf:47]: name conflict can happen on postgresqlserver only, not on database.
A server can host multiple databases.

File: examples/azure/terraform-azure-postgresql-example/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Terraform Azure PostgreSQL DB Example
+
+This folder contains a Terraform module that deploys resources in [Azure](https://azure.microsoft.com/) to demonstrate how you can use Terratest to write automated tests for your Azure Terraform code.
+This module deploys a database for PostgreSQL.
+
```
[the remark this answers is not in the document]
Comment (omichels) [reply examples/azure/terraform-azure-postgresql-example/README.md:5]: resolved

File: examples/azure/terraform-azure-postgresql-example/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+## Running automated tests against this module
+1. Sign up for [Azure](https://azure.microsoft.com/)
+1. Configure your Azure credentials using one of the [supported methods for Azure CLI
+   tools](https://docs.microsoft.com/en-us/cli/azure/azure-cli-configuration?view=azure-cli-latest)
+1. Install [Terraform](https://www.terraform.io/) and make sure it's on your `PATH`
+1. Configure your Terratest [Go test environment](../README.md) 
+1. `cd test/azure`
+1. `go build terraform_azure_mysqldb_example_test.go`
```
[the remark this answers is not in the document]
Comment (omichels) [reply examples/azure/terraform-azure-postgresql-example/README.md]: resolved

File: examples/azure/terraform-azure-postgresql-example/main.tf [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# ---------------------------------------------------------------------------------------------------------------------
+provider "azurerm" {
+  features {}
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A RESOURCE GROUP
+# ---------------------------------------------------------------------------------------------------------------------
+resource "azurerm_resource_group" "rg" {
+  name     = "postgresql-rg"
```
[the remark this answers is not in the document]
Comment (omichels) [reply examples/azure/terraform-azure-postgresql-example/main.tf]: resolved

File: examples/azure/terraform-azure-postgresql-example/main.tf [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  resource_group_name = azurerm_resource_group.rg.name
+
+  sku_name = "B_Gen5_2"
+
+  storage_mb                   = 5120
+  backup_retention_days        = 7
+  geo_redundant_backup_enabled = false
+  auto_grow_enabled            = true
+
+  administrator_login          = "pgsqladmin"
```
[the remark this answers is not in the document]
Comment (omichels) [reply examples/azure/terraform-azure-postgresql-example/main.tf:38]: resolved

File: examples/azure/terraform-azure-postgresql-example/output.tf [hunk 6]
```diff
@@ -0,0 +1,12 @@
+output "sku_name" {
+    value = azurerm_postgresql_server.postgresqlserver.sku_name
+}
+
+output "servername" {
+    value = azurerm_postgresql_server.postgresqlserver.name
```
[the remark this answers is not in the document]
Comment (omichels) [reply examples/azure/terraform-azure-postgresql-example/output.tf]: resolved

File: examples/azure/terraform-azure-postgresql-example/output.tf [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  
+}
+
+output "rgname" {
+    value = azurerm_resource_group.rg.name
+}
```
[the remark this answers is not in the document]
Comment (omichels) [reply examples/azure/terraform-azure-postgresql-example/output.tf:12]: used terraform fmt

File: examples/azure/terraform-azure-postgresql-example/main.tf [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+# CONFIGURE OUR AZURE CONNECTION
+# ---------------------------------------------------------------------------------------------------------------------
+provider "azurerm" {
+  features {}
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A RESOURCE GROUP
+# ---------------------------------------------------------------------------------------------------------------------
+resource "azurerm_resource_group" "rg" {
```
[the remark this answers is not in the document]
Comment (omichels) [reply examples/azure/terraform-azure-postgresql-example/main.tf:18]: resolved

File: examples/azure/terraform-azure-postgresql-example/output.tf [hunk 9]
```diff
@@ -0,0 +1,12 @@
+output "sku_name" {
+    value = azurerm_postgresql_server.postgresqlserver.sku_name
+}
+
+output "servername" {
+    value = azurerm_postgresql_server.postgresqlserver.name
+  
+}
+
+output "rgname" {
```
[the remark this answers is not in the document]
Comment (omichels) [reply examples/azure/terraform-azure-postgresql-example/output.tf:10]: resolved

**Comment by Jfmask** [comment]
Hello.

This PR is very useful for us so I would like to ask, is there any chance that this functionality will be accepted?

**Review by HadwaAbdelhalem — APPROVED** [review]
Thanks @omichels . @ Jfmask yes, it is in final review phase .  @yorinasub17 this is ready for your review

File: modules/azure/postgresql.go [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+	// Attach authorizer to the client
+	postgresqlClient.Authorizer = *authorizer
+
+	return &postgresqlClient, nil
+}
+
+// GetPostgresqlServer is a helper function that gets the server.
+// This function would fail the test if there is an error.
+func GetPostgresqlServer(t testing.TestingT, resGroupName string, serverName string, subscriptionID string) *postgresql.Server {
```
Comment (yorinasub17) [inline modules/azure/postgresql.go:36]: Can you make all the functions consistent and use `PostgreSQL` instead of `Postgresql`?

**Review by yorinasub17 — COMMENTED** [review]
LGTM! Had one more update on function consistency, but should be able to merge once that is addressed!

File: modules/azure/postgresql.go (hunk 10 above)
Comment (HadwaAbdelhalem) [reply modules/azure/postgresql.go:36]: @omichels could you please update the PR to resolve the last comment, or give me access to your repo and I would do it.

Comment (omichels) [reply modules/azure/postgresql.go]: "all functions consistent using PostgreSQL" 
commit: 00727ae

**Review by HadwaAbdelhalem — APPROVED** [review]
Tested and approved, @yorinasub17  ready for your final check.

**Review by yorinasub17 — APPROVED** [review]
LGTM! Thanks for the contribution! Will merge and release now.

**Pull request merged**